lib/solidus_multi_domain/engine.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'spree/core'
+
+module SolidusMultiDomain
+  class Engine < Rails::Engine
+    include SolidusSupport::EngineExtensions::Decorators
+
+    isolate_namespace ::Spree
+
+    engine_name 'solidus_multi_domain'
+
+    # use rspec for tests
+    config.generators do |g|
+      g.test_framework :rspec
+    end
+
+    def self.activate
+      ::Spree::Config.searcher_class = ::Spree::Search::MultiDomain
+      ApplicationController.include SolidusMultiDomain::MultiDomainHelpers
+    end
+
+    config.to_prepare &method(:activate).to_proc
+
+    initializer 'spree.promo.register.promotions.rules' do |app|
+      app.config.spree.promotions.rules << ::Spree::Promotion::Rules::Store
+    end
+
+    def self.admin_available?
```

Comment:
Who is using these methods?

Reply:
The extension only decorates some classes if the corresponding framework is available

<img width="694" alt="image" src="https://user-images.githubusercontent.com/1051/73643044-11151880-4673-11ea-803e-856f2135e8a9.png">

Reply:
Got it. We already have the same methods in [`solidus_support`](https://github.com/solidusio/solidus_support/blob/master/lib/solidus_support.rb#L52), which is a dependency already. 

Anyway, I don't think it is the right approach. 

We created this [`solidus_support`](https://github.com/solidusio/solidus_support/blob/cd4892d959d49f557bf13832c68f64dfcb113746/lib/solidus_support/engine_extensions.rb) module. Including that module will be just a matter of adding these decorators in the "right" folder, as done [here](https://github.com/solidusio/solidus_auth_devise/tree/master/lib/decorators). 

The module is not yet been released (we are waiting for some test in real extensions) so you can use it only by using the github version for now. You can make a test to see if everything works correctly or we can just switch to the `solidus_support` `*_available?` methods and face this change in a future PR, up to you!

Reply:
This is the PR for `solidus_auth_devise` to start using this approach: https://github.com/solidusio/solidus_auth_devise/pull/182

Reply:
I would prefer not to mix the two changes, after all this PR is targeted at just upgrading the files to the latest prescriptions of solidus_dev_support and make the specs pass. 
I opened a separate issue as you suggested to keep track of this, and possibly use this extension as the testbed to validate that solidus_support PR

https://github.com/solidusio-contrib/solidus_multi_domain/issues/139